tests/integration/__init__.py

```diff
@@ -389,6 +394,51 @@ def _id_and_mark(f):
 requires_custom_payload = pytest.mark.skipif(SCYLLA_VERSION is not None or PROTOCOL_VERSION < 4,
                                             reason='Scylla does not support custom payloads. Cassandra requires native protocol v4.0+')
 xfail_scylla = lambda reason, *args, **kwargs: pytest.mark.xfail(SCYLLA_VERSION is not None, reason=reason, *args, **kwargs)
+
+
+def xfail_scylla_version_lt(reason, oss_scylla_version, ent_scylla_version, *args, **kwargs):
+    """
+    It is used to mark tests that are going to fail on certain scylla versions.
+
+    ent_scylla_version should contain at least first major: `x.1.1` or `x.1` or `x`
+    """
+    if not reason.startswith("scylladb/scylladb#"):
+        raise ValueError('reason should start with scylladb/scylladb#<issue-id> to reference issue in scylla repo')
+
+    if not isinstance(ent_scylla_version, (tuple, list)):
+        ent_scylla_version = (ent_scylla_version,)
+    first_major = None
+    ent_scylla_version = [Version(v) for v in ent_scylla_version]
+
```

Comment:
Actually it looks like `Version` only accepts string / bytes:
```
>>> from packaging.version import Version
>>> repr(Version((1, 2, 3)))
```

so I have no idea what the first variant is supposed to do, as it passes tuple and list to Version

Reply:
> I'll write my comments as I try to understand this function - hopefully it will be at least a bit helpful with improving comments / changing code.
> 
> So here either `ent_scylla_version` was `(tuple, list)` - so this will be a [Version(tuple), Version(list)]`, or it was changed to 1-element tuple, so it will be `Version(tuple)`- I assume this is the case for passing a string like`2023.1`, but I don't get what first variant is for. Why do we have an array of versions???

Answer is up here https://github.com/scylladb/python-driver/pull/363#discussion_r1835704687